src/App.tsx

```diff
@@ -1,223 +1,294 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos,
+  createTodos,
+  deleteTodos,
+  UpdateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoLIst } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { Header } from './components/Header';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [errorAdd, setErrorAdd] = useState(false);
+  const [errorRemove, setErrorRemove] = useState(false);
+  const [errorUpdate, setErrorUpdate] = useState(false);
+  const [errorEmptyTitile, setErrorEmptyTitile] = useState(false);
```

Comment:
let's create one variable in state for setting errors, and will use enum with available options to define which error was occured? in this way for handling one more error you'll need only to add and process one more option in enum, instead of creating new variable in state and add code for it's processing